waku/waku_store/common.nim

```diff
@@ -23,17 +23,19 @@ const
 type WakuStoreResult*[T] = Result[T, string]
 
 
-## Waku message digest
+## Waku message hash
 
-type MessageDigest* = MDigest[256]
+type MessageHash* = MDigest[256]
 
-proc computeDigest*(msg: WakuMessage): MessageDigest =
+proc computeDigest*(msg: WakuMessage, pubSubTopic: string): MessageHash =
```

Comment:
This should not be what's used to compute the message hash - this function exists for various historical reasons that I can't fully remember. It must indeed eventually be deprecated, but the authoritative message digest computation is: https://github.com/waku-org/nwaku/blob/master/waku/waku_core/message/digest.nim

Reply:
In other words, changing this function is unnecessarily going to break backwards compatibility. We simply want to add the message digest as defined in the RFC and implemented in https://github.com/waku-org/nwaku/blob/master/waku/waku_core/message/digest.nim as the primary key index column. This will allow us to extend the Store protocol in future to operate on these unique message digests. These will also replace our current cursors, but we'll do so in a way that doesn't break backwards compatbility. IMO for now the existing query behaviour, and especially cursors, shouldn't change as it will be very high-impact, unless I'm missing something?